examples/atari/atari_dqn_wandb.py

```diff
@@ -0,0 +1,216 @@
+    return wrap_deepmind(
+        args.task,
+        frame_stack=args.frames_stack,
+        episode_life=False,
+        clip_rewards=False
+    )
+
+
+def test_dqn(args=get_args()):
+    env = make_atari_env(args)
+    args.state_shape = env.observation_space.shape or env.observation_space.n
+    args.action_shape = env.action_space.shape or env.action_space.n
+    # should be N_FRAMES x H x W
+    print("Observations shape:", args.state_shape)
+    print("Actions shape:", args.action_shape)
+    # make environments
+    train_envs = SubprocVectorEnv(
+        [lambda: make_atari_env(args) for _ in range(args.training_num)]
+    )
+    test_envs = SubprocVectorEnv(
+        [lambda: make_atari_env_watch(args) for _ in range(args.test_num)]
+    )
+    # seed
+    np.random.seed(args.seed)
+    torch.manual_seed(args.seed)
+    train_envs.seed(args.seed)
+    test_envs.seed(args.seed)
+    # define model
+    net = DQN(*args.state_shape, args.action_shape, args.device).to(args.device)
+    optim = torch.optim.Adam(net.parameters(), lr=args.lr)
+    # define policy
+    policy = DQNPolicy(
+        net,
+        optim,
+        args.gamma,
+        args.n_step,
+        target_update_freq=args.target_update_freq
+    )
+    # load a previous policy
+    log_path = os.path.join(args.logdir, args.task, 'dqn')
+    logger = WandbLogger(
+        save_interval=1, project='atari', name='test_resume', run_id=args.resume_id
+    )
+    if args.resume_id:
+        logger.restore_data()
+        ckpt_path = os.path.join(log_path, 'checkpoint.pth')
+        checkpoint = torch.load(ckpt_path, map_location=args.device)
+        policy.load_state_dict(checkpoint['model'])
+        print("Loaded agent from: ", log_path)
+
+    # replay buffer: `save_last_obs` and `stack_num` can be removed together
+    # when you have enough RAM
+    buffer = VectorReplayBuffer(
+        args.buffer_size,
+        buffer_num=len(train_envs),
+        ignore_obs_next=True,
+        save_only_last_obs=True,
+        stack_num=args.frames_stack
+    )
+    # collector
+    train_collector = Collector(policy, train_envs, buffer, exploration_noise=True)
+    test_collector = Collector(policy, test_envs, exploration_noise=True)
+
+    def save_fn(policy):
+        torch.save(policy.state_dict(), os.path.join(log_path, 'policy.pth'))
+
+    def stop_fn(mean_rewards):
+        if env.spec.reward_threshold:
+            return mean_rewards >= env.spec.reward_threshold
+        elif 'Pong' in args.task:
+            return mean_rewards >= 20
+        else:
+            return False
+
+    def train_fn(epoch, env_step):
+        # nature DQN setting, linear decay in the first 1M steps
+        if env_step <= 1e6:
+            eps = args.eps_train - env_step / 1e6 * \
+                (args.eps_train - args.eps_train_final)
+        else:
+            eps = args.eps_train_final
+        policy.set_eps(eps)
+        logger.write('train/eps', env_step, eps)
```

Comment:
this is the root course of that bug, I'll fix that

Reply:
thanks

Reply:
#444